core/controllers/contributor_dashboard.py

```diff
@@ -308,25 +338,32 @@ def get(self):
 
         exp = exp_fetchers.get_exploration_by_id(exp_id, strict=False)
         if exp is None:
-            raise self.PageNotFoundException('Invalid exp_id: %s' % exp_id)
+            raise self.PageNotFoundException()
         state_names_to_content_id_mapping = exp.get_translatable_text(
             target_language_code)
-        if not state_names_to_content_id_mapping.has_key(state_name):
-            raise self.PageNotFoundException(
-                'Invalid state_name: %s' % state_name)
-        content_id_mapping = state_names_to_content_id_mapping[state_name]
+        if state_name not in state_names_to_content_id_mapping:
+            raise self.PageNotFoundException()
+        content_id_to_text_mapping = (
+            state_names_to_content_id_mapping[state_name])
         translated_texts = {}
+        errors = {}
         for content_id in content_ids:
-            if content_id_mapping.has_key(content_id):
-                source_text = content_id_mapping[content_id]
+            try:
+                if content_id not in content_id_to_text_mapping:
+                    raise LookupError('Invalid content_id: %s' % content_id)
```

Comment:
Anything raised at this level should be self.InvalidInputException or self.PageNotFoundException etc., following the usual conventions for 400, 404, 401. In this case I think it's a 400.

Reply:
Wouldn't returning an exception here prevent a partially successful translation from returning the successful translations?

Reply:
Fixed. See comments below.